attempt to also run e2e tests that needs gpus

winglian opened the pull request.

winglian committed:
attempt to also run e2e tests that needs gpus

hamelsmu commented:
Hopefully this won't block CI pipeline too much

hamelsmu reviewed (changes requested):
Looks like something is wrong with this command see the CI logs

winglian commented:
> Looks like something is wrong with this command see the CI logs

`'"all"` 🤦

hamelsmu commented on .github/workflows/tests-docker.yml:
```diff
@@ -43,4 +43,4 @@ jobs:
       - name: Unit Tests w docker image
         run: |
-          docker run --rm test-axolotl pytest --ignore=tests/e2e/ /workspace/axolotl/tests/
+          docker run --privileged --gpus '"all" --rm test-axolotl pytest /workspace/axolotl/tests/
```
You have an extra single quote here

winglian committed:
fix stray quote

hamelsmu commented:
@winglian I dont have access to the hosted runner, but I think that we need to set nvidia runtime as the default (just trying to interpret the error message).  

On your self-hosted runner, in `/etc/docker/daemon.json` the `default-runtime` should be this:

```json
{
  "default-runtime": "nvidia",
  "runtimes": {
    "nvidia": {
      "path": "nvidia-container-runtime",
      "runtimeArgs": []
    }
  }
}
```

If this is not the case, make the change and then restart docker by running

```
sudo systemctl restart docker
```

winglian committed:
checkout specific github ref

winglian committed:
dockerfile for tests with proper checkout

ensure wandb is dissabled for docker pytests
clear wandb env after testing
clear wandb env after testing
make sure to provide a default val for pop
tryin skipping wandb validation tests
explicitly disable wandb in the e2e tests
explicitly report_to None to see if that fixes the docker e2e tests
split gpu from non-gpu unit tests
skip bf16 check in test for now
build docker w/o cache since it uses branch name ref
revert some changes now that caching is fixed
skip bf16 check if on gpu w support

winglian committed:
pytest skip for auto-gptq requirements

winglian committed:
skip mamba tests for now, split multipack and non packed lora llama tests

winglian committed:
split tests that use monkeypatches

winglian committed:
fix relative import for prev commit

winglian committed:
move other tests using monkeypatches to the correct run
